packages/core/src/index.ts

```diff
@@ -0,0 +1,32 @@
+import { GlintConfig, loadConfig } from '@glint/config';
+import DocumentCache from './common/document-cache';
+import TransformManager from './common/transform-manager';
+import GlintLanguageServer from './language-server/glint-language-server';
+
+/** @internal */
+export interface ProjectAnalysis {
+  glintConfig: GlintConfig;
+  transformManager: TransformManager;
+  languageServer: GlintLanguageServer;
+  shutdown: () => void;
+}
+
+/**
+ * This function is available to consumers as an unstable API. We will not go
+ * out of our way to change or break it, but there may be breaking changes
+ * to its behavior or type signature outside of major version bumps.
+ *
+ * See the `auto-glint-nocheck` implementation in `@glint/scripts` for a
+ * sample use of this API.
+ *
+ * @internal
+ */
+export function analyzeProject(projectDirectory: string = process.cwd()): ProjectAnalysis {
+  let glintConfig = loadConfig(projectDirectory);
+  let documents = new DocumentCache(glintConfig);
+  let transformManager = new TransformManager(glintConfig, documents);
```

Comment:
A question, mostly for my own edification: is there any reason we would need this to use the `TransformManagerPool` approach, or is the idea that the things this would be used for would always have only a single manager corresponding to the single language server, regardless of the project structure?

If we *would* want to support a flow that needed the pool, it's also totally fine to add later, too; but I don't have a strong intuition about whether that would even make sense—I think *no*?

Reply:
For the sole use case we currently have, I think saying "run `auto-glint-nocheck` for each individual project as you turn on Glint for it" makes sense, but I could definitely imagine other programmatic use cases that might want something like `analyzeCompositeProject` that would provide a more holistic view of a composite setup.

My inclination is to hold off until we have a concrete use case for that in mind, as I'm _guessing_ the shape we'd want to return from that might be slightly different.

Reply:
Makes sense to me!